Makefile_packaging.mk

```diff
@@ -0,0 +1,241 @@
+	    if ! grep -q "^Origin:\|^Author:\|^From:" "$$f" ;then \
+	      sed -i '1 iOrigin: other' \
+	        "$(DEB_BUILD)/debian/patches/$$fb" ;fi ; \
+	  fi ; \
+	done
+	touch $@
+
+
+# Move the debian files into the Debian directory.
+ifeq ($(ID_LIKE),debian)
+$(DEB_TOP)/.deb_files : $(shell find debian -type f) \
+	  $(DEB_TOP)/.detar | \
+	  $(DEB_BUILD)/debian/
+	find debian -maxdepth 1 -type f -exec cp '{}' '$(DEB_BUILD)/{}' ';'
+	if [ -e debian/source ]; then \
+	  cp -r debian/source $(DEB_BUILD)/debian; fi
+	if [ -e debian/local ]; then \
+	  cp -r debian/local $(DEB_BUILD)/debian; fi
+	if [ -e debian/examples ]; then \
+	  cp -r debian/examples $(DEB_BUILD)/debian; fi
+	if [ -e debian/upstream ]; then \
+	  cp -r debian/upstream $(DEB_BUILD)/debian; fi
+	if [ -e debian/tests ]; then \
+	  cp -r debian/tests $(DEB_BUILD)/debian; fi
+	rm -f $(DEB_BUILD)/debian/*.ex $(DEB_BUILD)/debian/*.EX
+	rm -f $(DEB_BUILD)/debian/*.orig
+	touch $@
+endif
+
+# see https://stackoverflow.com/questions/2973445/ for why we subst
+# the "rpm" for "%" to effectively turn this into a multiple matching
+# target pattern rule
+$(subst rpm,%,$(RPMS)): $(SPEC) $(SOURCES)
+	rpmbuild -bb $(COMMON_RPM_ARGS) $(RPM_BUILD_OPTIONS) $(SPEC)
+
+$(subst deb,%,$(DEBS)): $(DEB_BUILD).tar.$(SRC_EXT) \
+	  $(DEB_TOP)/.deb_files $(DEB_TOP)/.detar $(DEB_TOP)/.patched
+	rm -f $(DEB_TOP)/*.deb $(DEB_TOP)/*.ddeb $(DEB_TOP)/*.dsc \
+	      $(DEB_TOP)/*.dsc $(DEB_TOP)/*.build* $(DEB_TOP)/*.changes \
+	      $(DEB_TOP)/*.debian.tar.*
+	rm -rf $(DEB_TOP)/*-tmp
+	cd $(DEB_BUILD); debuild --no-lintian -b -us -uc
+	cd $(DEB_BUILD); debuild -- clean
+	git status
+	rm -rf $(DEB_TOP)/$(NAME)-tmp
+	lfile1=$(shell echo $(DEB_TOP)/$(NAME)[0-9]*_$(DEB_VERS)-1_amd64.deb);\
+	  lfile=$$(ls $${lfile1}); \
+	  lfile2=$${lfile##*/}; lname=$${lfile2%%_*}; \
+	  dpkg-deb -R $${lfile} \
+	    $(DEB_TOP)/$(NAME)-tmp; \
+	  if [ -e $(DEB_TOP)/$(NAME)-tmp/DEBIAN/symbols ]; then \
+	    sed 's/$(DEB_RVERS)-1/$(DEB_BVERS)/' \
+	    $(DEB_TOP)/$(NAME)-tmp/DEBIAN/symbols \
+	    > $(DEB_BUILD)/debian/$${lname}.symbols; fi
+	cd $(DEB_BUILD); debuild -us -uc
+	rm $(DEB_BUILD).tar.$(SRC_EXT)
+	for f in $(DEB_TOP)/*.deb; do \
+	  echo $$f; dpkg -c $$f; done
+
+ifneq ($(SRC_EXT),rpm)
+$(SRPM): $(SPEC) $(SOURCES)
+	rpmbuild -bs $(COMMON_RPM_ARGS) $(SPEC)
+endif
+
+srpm: $(SRPM)
+
+$(RPMS): Makefile
+
+rpms: $(RPMS)
+
+$(DEBS): Makefile
+
+debs: $(DEBS)
+
+ls: $(TARGETS)
+	ls -ld $^
+
+mockbuild: $(SRPM) Makefile
+	mock $(MOCK_OPTIONS) $<
+
+rpmlint: $(SPEC)
+	rpmlint $<
+
+# Debian wants a distclean target
+#distclean:
+#	@echo "distclean"
+
+check-env:
+ifndef DEBEMAIL
+	$(error DEBEMAIL is undefined)
+endif
+ifndef DEBFULLNAME
+	$(error DEBFULLNAME is undefined)
+endif
+
+show_version:
+	@echo $(VERSION)
+
+show_tag:
+	@echo ${TAG}
+
+show_release:
+	@echo $(RELEASE)
+
+show_rpms:
+	@echo $(RPMS)
+
+show_source:
+	@echo $(SOURCE)
+	@echo -$(NAME)-$(VERSION).tar.$(SRC_EXT)-
+
+source: $(notdir $(SOURCE))
```

Comment:
I don't think we need this if we make `build` build the SRPM instead of the specfile.

Reply:
Then I need a way of getting the SRPM for a download, and if the SRPM needs patching, we are back to the reason this was switched to having the spec file.

Reply:
`make srpm` makes the SRPM from the repo.